autogpt_platform/backend/backend/blocks/google/gmail.py

```diff
@@ -0,0 +1,444 @@
+            categories={BlockCategory.COMMUNICATION},
+            input_schema=GmailAddLabelBlock.Input,
+            output_schema=GmailAddLabelBlock.Output,
+            disabled=not GOOGLE_OAUTH_IS_CONFIGURED,
+            test_input={
+                "message_id": "12345",
+                "label_name": "Important",
+                "credentials": TEST_CREDENTIALS_INPUT,
+            },
+            test_credentials=TEST_CREDENTIALS,
+            test_output=[
+                (
+                    "result",
+                    {"status": "Label added successfully", "label_id": "Label_1"},
+                ),
+            ],
+            test_mock={
+                "_add_label": lambda *args, **kwargs: {
+                    "status": "Label added successfully",
+                    "label_id": "Label_1",
+                },
+            },
+        )
+
+    def run(
+        self, input_data: Input, *, credentials: GoogleCredentials, **kwargs
+    ) -> BlockOutput:
+        try:
+            service = GmailReadBlock._build_service(credentials, **kwargs)
+            result = self._add_label(
+                service, input_data.message_id, input_data.label_name
+            )
+            yield "result", result
+        except Exception as e:
+            yield "error", str(e)
+
+    def _add_label(self, service, message_id: str, label_name: str) -> dict:
+        label_id = self._get_or_create_label(service, label_name)
+        service.users().messages().modify(
+            userId="me", id=message_id, body={"addLabelIds": [label_id]}
+        ).execute()
+        return {"status": "Label added successfully", "label_id": label_id}
+
+    def _get_or_create_label(self, service, label_name: str) -> str:
+        label_id = self._get_label_id(service, label_name)
+        if not label_id:
+            label = (
+                service.users()
+                .labels()
+                .create(userId="me", body={"name": label_name})
+                .execute()
+            )
+            label_id = label["id"]
+        return label_id
+
+    def _get_label_id(self, service, label_name: str) -> str | None:
+        results = service.users().labels().list(userId="me").execute()
+        labels = results.get("labels", [])
+        for label in labels:
+            if label["name"] == label_name:
+                return label["id"]
+        return None
+
+
+class GmailRemoveLabelBlock(Block):
+    class Input(BlockSchema):
+        credentials: GoogleCredentialsInput = GoogleCredentialsField(
+            ["https://www.googleapis.com/auth/gmail.modify"]
+        )
+        message_id: str = SchemaField(
+            description="Message ID to remove label from",
+        )
+        label_name: str = SchemaField(
+            description="Label name to remove",
+        )
+
+    class Output(BlockSchema):
+        result: dict = SchemaField(
+            description="Label removal result",
+        )
```

Comment:
is there anything useful in here besides "operation succeeded"?

Reply:
not really?

Reply:
This should be discussed further to find a good pattern to use for outputs like this